### PR TITLE
Add debug log statements for pinot PQL

### PR DESCRIPTION
**Pull request by naman-patel**
## Description
To do a better debuging of Pinot performance issues we are trying to log the queries. There is no way we can do that. I do feel there should be a general approach for enabling this across but for now we desperately need this.

## Additional context and related issues
The PR allows a connector setting to log the queries executed against a pinot catalog. This helps better query tracking.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/DynamicTablePqlExtractor.java [hunk 1]
```diff
@@ -92,7 +100,11 @@ public static String extractPql(DynamicTable table, TupleDomain<ColumnHandle> tu
             }
             builder.append(table.limit().getAsLong());
         }
-        return builder.toString();
+        String pql = builder.toString();
+        if (queryLoggingEnabled && log.isInfoEnabled()) {
+            log.info("Generated PQL: %s", pql);
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/DynamicTablePqlExtractor.java:105]: Please revert config changes. We can control log level with [log.properties](https://trino.io/docs/current/installation/deployment.html#log-levels).

Also, please change to debug level. Same for another place.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java [hunk 2]
```diff
@@ -81,7 +84,11 @@ public static String generatePql(PinotTableHandle tableHandle, List<PinotColumnH
         }
         pqlBuilder.append(" LIMIT ")
                 .append(limit);
-        return pqlBuilder.toString();
+        String pql = pqlBuilder.toString();
+        if (queryLoggingEnabled && log.isInfoEnabled()) {
+            log.info("Generated PQL: %s", pql);
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java:89]: Change to log.debug.

Comment (naman-patel) [reply plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java:89]: Debug logs are not enabled. If we want to see logs with single config its better to make it info. By default its false and you can enable / disable it per catalog.

Comment (ebyhr) [reply plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java:89]: Disagreed. We don't manage log settings in catalog files.

Comment (ebyhr) [reply plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java:89]: This can lead to a log flood. Please change to debug level.

**Comment by naman-patel** [comment]
@ebyhr done. Changed all to debug.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java (hunk 2 above)
Comment (naman-patel) [reply plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java]: I have made these debug. But the logs are not enabled by default. It has to be configured to be enabled. Also the current approach surely leads to bigger flood of logs as enabling the logger enables for all catalogs and there is no per catalog setting. On top of this enables debug for the entire class not just one log statement. But since you are the moderator I yield.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotPageSourceProvider.java [hunk 3]
```diff
@@ -74,12 +74,7 @@ public ConnectorPageSource createPageSource(
             handles.add((PinotColumnHandle) handle);
         }
         PinotTableHandle pinotTableHandle = (PinotTableHandle) tableHandle;
-        String query = generatePql(
-                pinotTableHandle,
-                handles,
-                pinotSplit.getSuffix(),
-                pinotSplit.getTimePredicate(),
-                limitForSegmentQueries);
+        String query = generatePql(pinotTableHandle, handles, pinotSplit.getSuffix(), pinotSplit.getTimePredicate(), limitForSegmentQueries);
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotPageSourceProvider.java:77]: Please squash commits into one.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/DynamicTablePqlExtractor.java [hunk 4]
```diff
@@ -92,7 +95,11 @@ public static String extractPql(DynamicTable table, TupleDomain<ColumnHandle> tu
             }
             builder.append(table.limit().getAsLong());
         }
-        return builder.toString();
+        String pql = builder.toString();
+        if (log.isDebugEnabled()) {
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/DynamicTablePqlExtractor.java:99]: No need to use this condition. Please just use `log.debug`.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/DynamicTablePqlExtractor.java [hunk 5]
```diff
@@ -92,7 +95,11 @@ public static String extractPql(DynamicTable table, TupleDomain<ColumnHandle> tu
...
+            log.debug("Generated PQL: %s", pql);
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/DynamicTablePqlExtractor.java:100]: The logging place seems wrong. We should log the generated PQL just before the execution.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java [hunk 6]
```diff
@@ -81,7 +84,11 @@ public static String generatePql(PinotTableHandle tableHandle, List<PinotColumnH
         }
         pqlBuilder.append(" LIMIT ")
                 .append(limit);
-        return pqlBuilder.toString();
+        String pql = pqlBuilder.toString();
+        if (log.isDebugEnabled()) {
+            log.debug("Generated PQL: %s", pql);
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java:89]: Note that we already have the following line:
https://github.com/trinodb/trino/blob/58ff0c0f919b4fd2c00656fb18de4e749c070c30/plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java#L552

**Comment by ebyhr** [comment]
Please update PR title.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java (hunk 6 above)
Comment (naman-patel) [reply plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java:89]: Are you saying this is not needed anymore?

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotPageSourceProvider.java (hunk 3 above)
Comment (naman-patel) [reply plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotPageSourceProvider.java:80]: @ebyhr done.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotPageSourceProvider.java [hunk 7]
```diff
@@ -75,7 +78,7 @@ public ConnectorPageSource createPageSource(
         }
         PinotTableHandle pinotTableHandle = (PinotTableHandle) tableHandle;
         String query = generatePql(pinotTableHandle, handles, pinotSplit.getSuffix(), pinotSplit.getTimePredicate(), limitForSegmentQueries);
-
+        LOG.debug("Pinot query: %s", query);
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotPageSourceProvider.java:81]: The logging place looks still wrong. Note that #22253 is going to move the above `generatePql` method call.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotPageSourceProvider.java [hunk 8]
```diff
@@ -94,7 +97,6 @@ public ConnectorPageSource createPageSource(
                 else {
                     pinotQueryInfo = new PinotQueryInfo(pinotTableHandle.getTableName(), query, 0);
                 }
-
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotPageSourceProvider.java:97]: Revert unrelated change.

**Comment by ebyhr** [comment]
> Log queries

Please use a more meaningful commit title.
